app/public/js/services/firebaseService.js

```diff
@@ -1,5 +1,9 @@
 angular.module('FirebaseService', []).service('firebaseService', function($rootScope) {
 
+  this.setUserPassword = function(newPassword) {
+    return $rootScope.user.updatePassword(newPassword);
```

Comment:
Maybe you should specify why it belongs in firebaseService. I do understand why but because I know how you coded, otherwise I wouldn't understand. Also, would it work the same way if you did it with firebase.auth().currentUser? If it did work, it would be more clear for people and you wouldn't be dependant on the way we store the user.